src/fmu/dataio/_model/fields.py

```diff
@@ -572,6 +583,44 @@ class FMUBase(BaseModel):
     See :class:`Model`."""
 
 
+class FMUIteration(FMUBase):
+    """
+    The ``fmu`` block contains all attributes specific to FMU. The idea is that the FMU
+    results data model can be applied to data from *other* sources - in which the
+    fmu-specific stuff may not make sense or be applicable.
+    This is a specialization of the FMU block for Iteration objects.
+    """
+
+    context: CaseContext
+    """The ``fmu.context`` block contains the FMU context in which this data object
+    was produced. See :class:`Context`.  """
+
+    iteration: Iteration
+    """The ``fmu.iteration`` block contains information about the iteration this data
+    object belongs to. See :class:`Iteration`. """
+
+
+class FMURealization(FMUBase):
+    """
+    The ``fmu`` block contains all attributes specific to FMU. The idea is that the FMU
+    results data model can be applied to data from *other* sources - in which the
+    fmu-specific stuff may not make sense or be applicable.
+    This is a specialization of the FMU block for Realization objects.
+    """
+
+    context: IterationContext
```

Comment:
I struggle a bit to understand why a realization would have the context `iteration` and not `realization` ? but might just be me that doesn't have the full context here of how this is going to be used 😄

Reply:
Our reasoning is:
A Realization is run as part of a specific Iteration. An Iteration can have many Realizations, so the Realizations are in the context of that Iteration.

Similar thought process for giving Iterations "CaseContext". A Case can have multiple Iterations, but an Iteration belongs to a specific case.

Reply:
It is probably fine 🙂  I just had the thought of `context=iteration` as data that were aggregated over an `iteration`.

Reply:
Any input on this @perolavsvendsen?

Reply:
Relevant, perhaps: https://github.com/equinor/fmu-dataio/issues/592

Reply:
And this: https://github.com/equinor/fmu-dataio/issues/293

Point is, I think I agree with @tnatt here that the `Iteration` (metadata) object should be fmu.context.stage=iteration (since it has an iteration.uuid) and that `Realization` (metadata) object should be fmu.context.stage=realization (since it has a realization.uuid).